ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java

```diff
@@ -16,112 +16,53 @@
   private static final long MAX_SEND_TIMEOUT_SECONDS = 5;
 
-  private final LongSupplier clock;
-
   public InsertValuesExecutor() {
-    this(System::currentTimeMillis);
+    super();
   }
 
   @VisibleForTesting
   InsertValuesExecutor(final LongSupplier clock) {
-    this.clock = Objects.requireNonNull(clock, "clock");
+    super(clock);
   }
 
-  public Optional<KsqlEntity> execute(
-      final ConfiguredStatement<InsertValues> statement,
-      final KsqlExecutionContext executionContext,
-      final ServiceContext serviceContext
+  protected void sendRecord(
+          final InsertValues insertValues,
+          final KsqlConfig config,
+          final KsqlExecutionContext executionContext,
+          final ServiceContext serviceContext
   ) {
-    if (!statement.getConfig().getBoolean(KsqlConfig.KSQL_INSERT_INTO_VALUES_ENABLED)) {
+    if (!config.getBoolean(KsqlConfig.KSQL_INSERT_INTO_VALUES_ENABLED)) {
       throw new KsqlException("The server has disabled INSERT INTO ... VALUES functionality. "
-          + "To enable it, restart your KSQL-server with 'ksql.insert.into.values.enabled'=true");
-    }
-
-    final InsertValues insertValues = statement.getStatement();
-    final DataSource<?> dataSource = executionContext
-        .getMetaStore()
-        .getSource(insertValues.getTarget().getSuffix());
-
-    if (dataSource == null) {
-      throw new KsqlException("Cannot insert values into an unknown stream/table: "
-          + insertValues.getTarget().getSuffix());
+              + "To enable it, restart your KSQL-server with"
+              + "'ksql.insert.into.values.enabled'=true");
     }
 
-    if (dataSource instanceof KsqlTable && ((KsqlTable<?>) dataSource).isWindowed()
-        || dataSource instanceof KsqlStream && ((KsqlStream<?>) dataSource).hasWindowedKey()) {
-      throw new KsqlException("Cannot insert values into windowed stream/table!");
-    }
-
-    final KsqlConfig config = statement.getConfig()
-        .cloneWithPropertyOverwrite(statement.getOverrides());
-
-    final RowData row = extractRow(insertValues, dataSource);
-    final byte[] key = serializeKey(row.key, dataSource);
-    final byte[] value = serializeRow(row.value, dataSource, config, serviceContext);
-
-    final String topicName = dataSource.getKafkaTopicName();
-
     // for now, just create a new producer each time
     final Producer<byte[], byte[]> producer = serviceContext
-        .getKafkaClientSupplier()
-        .getProducer(config.getProducerClientConfigProps());
+            .getKafkaClientSupplier()
+            .getProducer(config.getProducerClientConfigProps());
 
     final Future<RecordMetadata> producerCallResult = producer.send(
-        new ProducerRecord<>(
-            topicName,
-            null,
-            row.ts,
-            key,
-            value
-        )
-    );
+            buildProducerRecord(insertValues, config, executionContext, serviceContext));
```

Comment:
nit: it looks like we can call `buildProducerRecord` in the parent class, make that method private and pass then `ProducerRecord` in